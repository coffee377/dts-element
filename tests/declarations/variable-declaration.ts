import * as dts from '../index';

it('should return correctly with name', () => {
  expect(dts.emit(
    dts.create_variable_declaration({
      name: 'a',
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, export', () => {
  expect(dts.emit(
    dts.create_variable_declaration({
      name: 'a',
      export: true,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, type', () => {
  expect(dts.emit(
    dts.create_variable_declaration({
      name: 'a',
      type: dts.string_type,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, let', () => {
  expect(dts.emit(
    dts.create_variable_declaration({
      name: 'a',
      let: true,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, export, type, const', () => {
  expect(dts.emit(
    dts.create_variable_declaration({
      name: 'a',
      const: true,
      export: true,
      type: dts.null_type,
    }),
  )).toMatchSnapshot();
});
