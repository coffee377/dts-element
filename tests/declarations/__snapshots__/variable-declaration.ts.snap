// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly with name 1`] = `"var a: any;"`;

exports[`should return correctly with name, export 1`] = `"export var a: any;"`;

exports[`should return correctly with name, export, type, const 1`] = `"export const a: null;"`;

exports[`should return correctly with name, let 1`] = `"let a: any;"`;

exports[`should return correctly with name, type 1`] = `"var a: string;"`;
